_posts/jared_baker/2018-12-15-Bluestore.md

```diff
@@ -0,0 +1,171 @@
+ceph-disk zap <osd-id>
+
+# Mark the OSD as destroyed
+ceph osd destroy <osd-id> --yes-i-really-mean-it
+
+# Prepare the disk as Bluestore
+ceph-disk prepare --bluestore /dev/<block-device> --osd-id <osd-id>
+~~~
+
+Notice that when comparing the output of ceph osd tree before/after conversion that the class column may now indicate your osd type (ssd or hdd) which for us, was blank before.
+
+Filestore 'ceph osd tree'
+~~~bash
+ID   CLASS WEIGHT     TYPE NAME                STATUS REWEIGHT PRI-AFF
+  -1       8503.16504 root default
+ -84                0     rack 1
+  -5       1306.07776     rack rack1
+ -83        388.79956         host storage1-r1                               
+720         10.79999             osd.720         up  1.00000 1.00000
+721         10.79999             osd.721         up  1.00000 1.00000
+~~~
+
+Bluestore 'ceph osd tree'
+~~~bash
+ID   CLASS WEIGHT     TYPE NAME                STATUS REWEIGHT PRI-AFF
+  -1       8503.16504 root default
+ -84                0     rack 1
+  -5       1306.07776     rack rack1
+ -83        388.79956         host storage1-r1
+720   hdd   10.79999             osd.720         up  1.00000 1.00000
+721   hdd   10.79999             osd.721         up  1.00000 1.00000
+~~~
+
+### Filling
+Now it's time to populate the bluestore OSD's with data. Just like the draining step we will change the osd crush weight
+
+Example, to fill OSD's 720 to 755 (12TB disks) we would run:
+~~~bash
+for i in $(seq 720 755); do ceph osd crush reweight osd.$i 10.79999; done
+~~~
+
+Depending on the size of the disks you will need to find the right value for your osd crush weight. This is typically the size of the disk in TB. Read more about the differences of ceph osd reweight and ceph osd crush reweight [HERE](https://ceph.com/geen-categorie/difference-between-ceph-osd-reweight-and-ceph-osd-crush-reweight/)
+
+Wait for ceph health to return to HEALTH_OK.
+
+## Tracking and Monitoring
+We use a variety of tools to monitor and report on our OpenStack and Ceph cluster. For this migration we were able to monitor filestores xfs data partition and see the data be copied off to the rest of the cluster.
+<figure>
+    <img id="drain1" src="{{site.urlimg}}jared_baker/bluestore/drain1.png" data-featherlight="#drain1" />
+</figure>
+
+Once the osd is converted to bluestore we no longer have that trusty posix filesystem to monitor in Zabbix so we had to rely on output from ceph mgr to obtain how much data is living on the osd. Grafana and graphite has nice integration with ceph-mgr so that we are still able to see the granularity of data distribution on each storage node
+<figure>
+    <img id="zabbixvsgrafana" src="{{site.urlimg}}jared_baker/bluestore/zabbixvsgrafana.png" data-featherlight="#zabbixvsgrafana" />
+</figure>
+
+With 37 storage nodes to migrate and each one needing to drain, convert, fill, we had to track it somehow. We are a team of two so we set up a shared google spreadsheet and kept track of the progress.
```

Comment:
- Is the sheet private/internal OICR only?  Or available to public?  If public, would it be useful to link to the template so others can use?  (I know it's a simple spreadsheet anyone can do, but people tend to be lazy :))